src/libraries/System.Reflection.Emit/tests/TypeBuilder/TypeBuilderGetConstructor.cs

```diff
@@ -7,17 +7,6 @@ namespace System.Reflection.Emit.Tests
 {
     public class TypeBuilderGetConstructor
     {
-        [Fact]
-        public void GetConstructor_DeclaringTypeOfConstructorNotGenericTypeDefinition_ThrowsArgumentException()
```

Comment:
We should not be removing these tests. Instead we should keep them, and if they no longer throw, we should assert what they do is correct.

Reply:
That's the case from the commit f2efe72, but locally this implementation of these tests did pass, but it was failing on CI checks for Helix.
Could you please that commit and let me know what could be done with that?

Reply:
I'm not sure I follow your question.

> but it was failing on CI checks for Helix.

Was it just failing for mono? Maybe we need to fix/change mono as well?

Reply:
@eerhardt Sure, now I see it's Mono that is failing.
Can you somehow (even briefly) guide me through how to handle the change for Mono?

Reply:
The Mono specific code is located here:

https://github.com/dotnet/runtime/blob/2ac5e2bb8a48dcb3038307338a04914ba708edc2/src/mono/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.Mono.cs#L1840-L1865